src/jit/CallInl.h

```diff
@@ -94,40 +94,24 @@ static void emitCall(sljit_compiler* compiler, Instruction* instr)
     }
 
     Operand* operand = instr->operands();
-    for (auto it : functionType->param()) {
-        Operand dst = VARIABLE_SET(STACK_OFFSET(*stackOffset), Instruction::Offset);
-
-        switch (VARIABLE_TYPE(*operand)) {
-        case Instruction::ConstPtr:
-            ASSERT(!(VARIABLE_GET_IMM(*operand)->info() & Instruction::kKeepInstruction));
-            emitStoreImmediate(compiler, &dst, VARIABLE_GET_IMM(*operand), false);
-            break;
-        case Instruction::Register:
-            emitMove(compiler, Instruction::valueTypeToOperandType(it), operand, &dst);
-            break;
-        }
-
-        operand++;
-        stackOffset += (valueSize(it) + (sizeof(size_t) - 1)) / sizeof(size_t);
-    }
+    stackOffset = emitStoreOntoStack(compiler, operand, stackOffset, functionType->param(), true);
+    operand += instr->paramCount();
 
     if (instr->opcode() == ByteCode::CallIndirectOpcode) {
         CallIndirect* callIndirect = reinterpret_cast<CallIndirect*>(instr->byteCode());
 
-        switch (VARIABLE_TYPE(*operand)) {
+        switch (VARIABLE_TYPE(operand[-1])) {
```

Comment:
minor suggestion) IMHO using negative index could be unsafe (of course, in this case, the index is calculated correctly)
What about adjusting the `operand` beforehand like `operand--` to use positive index here?

Reply:
I have changed it. I often prefer to use negative indexing with immediate because is is supported by all cpus I know.

Reply:
Thanks for the quick update. 
(AFAIK negative indexing is not natural or standard in C++ when accessing array)